### PR TITLE
Fix: Serialize jobenv to envVars PS-15020

### PR DESCRIPTION
**Pull request by paperspace-philip**
Fixes serialization of job environment

Test Plan:
Create a job and specify jobEnv parameter. Use command "env" to see if job environment is affected

### CODE REVIEW

File: gradient/api_sdk/serializers/job.py [hunk 1]
```diff
@@ -1,9 +1,30 @@
...
+
+        return json.dumps(value)
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        try:
+            result = json.loads(value)
+            if not result:
+                return None
+        except error:
+            raise "Could not parse %s json" % attr
```
Comment (worksewell) [inline gradient/api_sdk/serializers/job.py:25]: Result never gets returned, not sure if this is actually used or just a placeholder function

**Comment by bbatha** [comment]
This seems to fix #322

**Comment by lefnire** [comment]
This worked for me! 
```
pip install git+https://github.com/Paperspace/gradient-cli.git@job-env
```
running same code from #322, where previously didn't

File: gradient/cli/jobs.py [hunk 2]
```diff
@@ -169,7 +169,14 @@ def common_jobs_create_options(f):
             "--jobEnv",
             "job_env",
             type=json_string,
-            help="Environmental variables ",
+            help="Environmental variables (deprecated)",
+            cls=common.GradientOption,
+        ),
```
Comment (worksewell) [inline gradient/cli/jobs.py:174]: I'd probably just remove this. If you want to support the old flag you can just add it to the `--envVars` option.

File: gradient/api_sdk/clients/job_client.py [hunk 3]
```diff
@@ -75,7 +76,7 @@ def create(
                 name='Example job',
                 command='pip install -r requirements.txt && python mnist.py',
                 ports='5000:5000',
-                job_env={
+                env_vars={
                     'CUSTOM_ENV'='Some value that will be set as system environment',
```
Comment (worksewell) [inline gradient/api_sdk/clients/job_client.py:80]: nit: this isn't valid python, we should change it to something like (or whatever):

```
env_vars={'CUSTOM_ENV': 'my value'}
```

File: gradient/api_sdk/clients/job_client.py [hunk 4]
```diff
@@ -113,7 +114,8 @@ def create(
         :param str working_directory: location of code to run. By default ``/paperspace``
         :param str experiment_id: Id of experiment to which job should be connected. If not provided there will be
             created new experiment for this job.
-        :param dict job_env: key value collection of envs that are used in code
+        :param dict job_env: key value collection of envs that are used in code (deprecated)
```
Comment (worksewell) [inline gradient/api_sdk/clients/job_client.py:117]: I'd just remove this

**Review by worksewell — APPROVED** [review]
Once the tests pass lgtm

**Pull request merged**